source/mir/ndslice/slice.d

```diff
@@ -583,6 +491,195 @@ pure nothrow unittest
 }
 
 /++
+Creates an array and an n-dimensional slice over it.
+Params:
+    lengths = list of lengths for each dimension
+Returns:
+    n-dimensional slice
++/
+Slice!(Lengths.length, Select!(replaceArrayWithPointer, T*, T[]))
+createSlice(T,
+    Flag!`replaceArrayWithPointer` replaceArrayWithPointer = Yes.replaceArrayWithPointer,
+    Lengths...)(Lengths lengths)
+    if (allSatisfy!(isIndex, Lengths) && Lengths.length)
+{
+    return .createSlice!(T, replaceArrayWithPointer)([lengths]);
+}
+
+/// ditto
+Slice!(N, Select!(replaceArrayWithPointer, T*, T[]))
+createSlice(T,
+    Flag!`replaceArrayWithPointer` replaceArrayWithPointer = Yes.replaceArrayWithPointer,
+    size_t N)(auto ref in size_t[N] lengths)
+{
+    immutable len = lengthsProduct(lengths);
+    assert(len, "Cannot create empty slice");
+    return new T[len].sliced!replaceArrayWithPointer(lengths);
+}
+
+///
+pure nothrow unittest
+{
+    auto slice = createSlice!int(5, 6, 7);
+    assert(slice.length == 5);
+    assert(slice.elementsCount == 5 * 6 * 7);
+    static assert(is(typeof(slice) == Slice!(3, int*)));
+
+    auto duplicate = createSlice!int(slice.shape);
+    duplicate[] = slice;
+}
+
+/++
+Allocates an array through a specified allocator and creates an n-dimensional slice over it.
+See also $(LINK2 std_experimental_allocator.html, std.experimental.allocator).
+Params:
+    alloc = allocator
+    lengths = list of lengths for each dimension
+    init = default value for array initialization
+Returns:
+    a structure with fields `array` and `slice`
++/
+auto makeSlice(T,
+    Flag!`replaceArrayWithPointer` replaceArrayWithPointer = Yes.replaceArrayWithPointer,
+    Allocator,
```

Comment:
something that I was asking myself when working on the combinatorics module - is there any  template check for the Allocator api? If not - we might want to create one.